src/types.tsx

```diff
@@ -0,0 +1,35 @@
+import { RouteProp } from "@react-navigation/native";
+import { StackNavigationProp } from "@react-navigation/stack";
+
+// Add any parameters that are included with the page
+type RootStackParamList = {
+  Login: undefined;
+  CreateAccount: undefined;
+};
+
+// Login Page Types
+
+type LoginPageRouteProp = RouteProp<RootStackParamList, "Login">;
+
+type LoginPageNavigationProp = StackNavigationProp<RootStackParamList, "Login">;
+
+export type LoginPageProps = {
+  route: LoginPageRouteProp;
+  navigation: LoginPageNavigationProp;
+};
+
+// Create Account Page Types
+type CreateAccountPageRouteProp = RouteProp<
+  RootStackParamList,
+  "CreateAccount"
+>;
+
+type CreateAccountPageNavigationProp = StackNavigationProp<
+  RootStackParamList,
+  "CreateAccount"
+>;
+
+export type CreateAccountPageProps = {
+  route: CreateAccountPageRouteProp;
+  navigation: CreateAccountPageNavigationProp;
+};
```

Comment:
Instead of a global types file can we try and keep everything local to where they are used. I don't think majority of the components / pages need to access the other components props. as for the routes types can we make a global file called routes? I think this structure will be easier for us as developers. Any thoughts @smit3164 ?

Reply:
I agree with keeping a majority of types local to their necessary files. I do think though as we add in types for the models we retrieve from the backend, we will need a few global types since they likely will be used in many places. Also, I think having a routes file will be helpful as we start adding more pages, so that there aren't random route names or types floating around. I'll go ahead and make some changes

Reply:
I agree that we will need some form of global types file, I personally think we should try and organize our globals by type, so keep one dedicated file for routes, and one for any database models (i usually call this one `models`). I think that should cover the basic global types we need for now right?

Reply:
I'm with Stephen on this one, but let's avoid being too granular as we progress - usually that only causes headaches.